### PR TITLE
✨  Feature/fwf 3305 default filter

### PR DESCRIPTION
**Pull request by shuhaib-aot**
# Issue Tracking

JIRA:
- https://aottech.atlassian.net/browse/FWF-3305
- https://aottech.atlassian.net/browse/FWF-3306
Issue Type: BUG/ FEATURE

# Changes
- Added user data row for each tenant in db
- added `user/default-filter` endpoint for updating default filter
- created user table
-  added toggle button for default filter
- created alembic file
- set the default filter as the first filter in the filter list UI
- changed create new filter button to button-outline
 
### changed label to `set as default filter` but not updated in snap shot 
image

### CODE REVIEW

File: forms-flow-api/src/formsflow_api/schemas/user.py [hunk 1]
```diff
@@ -60,3 +60,17 @@ class Meta:  # pylint: disable=too-few-public-methods
...
+
+    class Meta:  # pylint: disable=too-few-public-methods
+        """Exclude unknown fields in the deserialized output."""
+
+        unknown = EXCLUDE
+
+    default_filter = fields.Int(data_key="defaultFilter", allow_none=True)
+    locale = fields.Str(data_key="locale")
+    user_name = fields.Str(data_key="userName", dump_only=True)
+    # tenant = fields.Str(data_key="tenantKey", dump_only=True)
```
Comment (auslin-aot) [inline forms-flow-api/src/formsflow_api/schemas/user.py:76]: whether tenant is needed in schema?

Comment (shuhaib-aot) [reply forms-flow-api/src/formsflow_api/schemas/user.py:76]: Actually I added for future use

File: forms-flow-web/src/containers/TaskHead.js [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 import ServiceFlowFilterListDropDown from "../components/ServiceFlow/filter/ServiceTaskFilterListDropDown";
 import { MULTITENANCY_ENABLED } from "../constants/constants";
-import { setBPMFilterLoader, setBPMFiltersAndCount, setSelectedTaskID, setViewType } from '../actions/bpmTaskActions';
+import { setBPMFilterLoader, setBPMFiltersAndCount, setDefaultFilter, setSelectedTaskID, setViewType } from '../actions/bpmTaskActions';
 import CreateNewFilterDrawer from "../components/ServiceFlow/list/sort/CreateNewFilter";
 import { useTranslation } from "react-i18next"; 
 import { fetchBPMTaskCount } from "../apiManager/services/bpmTaskServices";
-
+import { toast } from "react-toastify";
+import { updateDifaultFilter } from "../apiManager/services/userservices";
```
Comment (auslin-aot) [inline forms-flow-web/src/containers/TaskHead.js:12]: updateDefaultFilter typo

File: forms-flow-api/tests/unit/api/test_user.py [hunk 3]
```diff
@@ -68,3 +74,23 @@ def test_keycloak_users_list_invalid_group(app, client, session, jwt):
     headers = {"Authorization": f"Bearer {token}", "content-type": "application/json"}
     rv = client.get("/user?memberOfGroup=test123", headers=headers)
     assert rv.status_code == 400
+
+
+def test_default_fitler(app, client, session, jwt):
```
Comment (auslin-aot) [inline forms-flow-api/tests/unit/api/test_user.py:79]: please check test_default_filter typo?

**Review by auslin-aot — APPROVED** [review]

**Review by sumesh-aot — APPROVED** [review]

File: forms-flow-api/src/formsflow_api/services/user.py [hunk 4]
```diff
@@ -84,3 +89,20 @@ def paginate(self, data, page_number, page_size):
...
+        """Update user data."""
+        user: UserContext = kwargs["user"]
+        user_data = User.get_user_by_user_name(user_name=user.user_name)
+        if user_data:
+            if user_data.tenant is None and user.tenant_key:
+                data["tenant"] = user.tenant_key
+            user_data.update(data)
+        else:
+            data["user_name"] = user.user_name
+            data["tenant"] = user.tenant_key if user.tenant_key else None
```
[the remark this answers is not in the document]
Comment (shuhaib-aot) [reply forms-flow-api/src/formsflow_api/services/user.py:105]: okay

**Pull request merged**